src/main/kotlin/io/github/mojira/arisa/modules/RemoveTriagedMeqsModule.kt

```diff
@@ -4,38 +4,44 @@ import arrow.core.Either
 import arrow.core.extensions.fx
 import arrow.core.left
 import arrow.core.right
-import arrow.syntax.function.partially2
-import net.rcarz.jiraclient.Comment
-
-data class RemoveTriagedMeqsModuleRequest(
-    val priority: String?,
-    val triagedTime: String?,
-    val comments: List<Comment>
-)
+import arrow.syntax.function.partially1
 
 class RemoveTriagedMeqsModule(
-    val updateComment: (comment: Comment, body: String) -> Either<Throwable, Unit>,
-    val meqsTags: List<String>,
-    val removalReason: String
-) : Module<RemoveTriagedMeqsModuleRequest> {
-    override fun invoke(request: RemoveTriagedMeqsModuleRequest): Either<ModuleError, ModuleResponse> = Either.fx {
-        val meqsComments = request.comments.filter(::hasMeqsTag.partially2(meqsTags))
+    private val meqsTags: List<String>,
+    private val removalReason: String?
+) : Module<RemoveTriagedMeqsModule.Request> {
+    data class Comment(
+        val body: String,
+        val update: (String) -> Either<Throwable, Unit>
+    )
+
+    data class Request(
+        val priority: String?,
+        val triagedTime: String?,
+        val comments: List<Comment>
+    )
+
+    override fun invoke(request: Request): Either<ModuleError, ModuleResponse> = Either.fx {
         assertTriaged(request.priority, request.triagedTime).bind()
-        assertNotEmpty(meqsComments).bind()
 
-        tryRunAll({ updateComment(it, removeMeqsTags(it, meqsTags)) }, meqsComments).bind()
+        val updateMeqsComments = request.comments
+            .filter { hasMeqsTag(it.body) }
+            .map { it.update.partially1(removeMeqsTags(it.body)) }
+        assertNotEmpty(updateMeqsComments).bind()
+
+        tryRunAll(updateMeqsComments).bind()
     }
 
-    private fun hasMeqsTag(comment: Comment, meqsTags: List<String>) =
-        meqsTags.any { comment.body.contains(it) }
+    private fun hasMeqsTag(comment: String) =
+        meqsTags.any { comment.contains(it) }
 
-    private fun removeMeqsTags(comment: Comment, meqsTags: List<String>): String {
+    private fun removeMeqsTags(comment: String): String {
         val regex = (
             "MEQS(" +
             meqsTags.joinToString("|") { it.replace("MEQS", "") } +
             ")"
         ).toRegex()
-        return regex.replace(comment.body) { "MEQS_ARISA_REMOVED${it.groupValues[1]} Removal Reason: $removalReason" }
+        return regex.replace(comment) { "MEQS_ARISA_REMOVED${it.groupValues[1]}${ if (removalReason != null) " Removal Reason: $removalReason" else "" }" }
```

Comment:
The not having it as null would make things like this nicer :D

Reply:
Well then we would have `Removal Reason: <blank>` when non is specified 😉

Reply:
I can set it to that if you want, just seemed a little odd to me